src/SmartComponents/BaselinesPage/EditBaseline/helpers.js

```diff
@@ -1,49 +1,155 @@
+                if (subFact.name !== originalSubFact.name && subFact.value !== originalSubFact.value) {
+                    subFacts.push(subFact);
+                }
+            });
+            subFacts.push(newSubFact);
+            newAPIBody.values = subFacts;
+        } else {
+            newAPIBody = { name: factName, value: valueName };
+        }
     }
 
-    return childRows;
+    return newAPIBody;
 }
 
-function makeAPIPatch(rowData, originalAPIBody) {
-    if (rowData === undefined || originalAPIBody === undefined) {
+function buildDeletedSubFact(factToDelete, fact) {
+    let newParentFact;
+    let newSubFacts = [];
+
+    fact.values.forEach(function(subFact) {
+        if (subFact.name !== factToDelete.name && subFact.value !== factToDelete.value) {
+            newSubFacts.push(subFact);
+        }
+    });
+
+    newParentFact = { name: fact.name, values: newSubFacts };
+
+    return newParentFact;
+}
+
+function makeDeleteFactPatch(factToDelete, originalAPIBody, parentFactData) {
+    if (factToDelete === undefined || originalAPIBody === undefined) {
         return {};
     }
 
+    let patchBody = makeAPIPatch(factToDelete, originalAPIBody, parentFactData);
+    if (factToDelete.values) {
+        patchBody.push(factToDelete);
+    }
+
+    let patch = makePatchBody(patchBody, originalAPIBody.baseline_facts);
+
     /*eslint-disable camelcase*/
-    let patchBody = [];
-    let newAPIBody = { display_name: originalAPIBody.display_name };
-    originalAPIBody.baseline_facts.forEach(function(fact) {
-        if (fact.name !== rowData.name) {
-            patchBody.push(fact);
-        }
-    });
+    let newAPIBody = { display_name: originalAPIBody.display_name, facts_patch: patch };
+    /*eslint-enable camelcase*/
+
+    return newAPIBody;
+}
+
+function makeAddFactPatch(newFactData, originalAPIBody, oldFactData) {
+    if (newFactData === undefined || originalAPIBody === undefined) {
+        return {};
+    }
 
-    patchBody.push(rowData);
+    let patchBody = makeAPIPatch(newFactData, originalAPIBody, oldFactData);
+
+    patchBody.push(newFactData);
 
     let patch = makePatchBody(patchBody, originalAPIBody.baseline_facts);
 
-    newAPIBody.facts_patch = patch;
+    /*eslint-disable camelcase*/
+    let newAPIBody = { display_name: originalAPIBody.display_name, facts_patch: patch };
     /*eslint-enable camelcase*/
+
     return newAPIBody;
 }
 
+function makeAPIPatch(data, originalAPIBody, originalParentFact) {
+    let patchBody = [];
+
+    if (originalParentFact.length === 0) {
+        originalAPIBody.baseline_facts.forEach(function(fact) {
+            if (fact.name !== data.name) {
+                patchBody.push(fact);
+            }
+        });
+    } else {
+        originalAPIBody.baseline_facts.forEach(function(fact) {
+            if (fact.name !== originalParentFact.name) {
+                patchBody.push(fact);
+            }
+        });
+    }
```

Comment:
I wonder if this could be simplified to:

```
        originalAPIBody.baseline_facts.forEach(function(fact) {
            if (fact.name !== originalParentFact.name && originalParentFact.length > 0) || (fact.name !== data.name && originalParentFactLength === 0) {
                patchBody.push(fact);
           }
        });
```